WordPress/gutenberg
# Restore the navigation menus in the site editor

## What?

In #46995 the experimental flag for the Navigation Off Canvas editor has been removed but one check has been left out in the code base causing the navigation menu to not appear in the sidebar of the site editor.

I'm not removing all the remaining artifacts of the experimental flag which fixes the site editor sidebar as well. This PR also removes the navigation sidebar within the edit mode as it's now replaced with the one in the browse mode sidebar.

## Testing Instructions

1- Check that the navigation menu item appears in the left sidebar of the site editor but not within the edit mode.
